src/diagonal.works/b6/world.go

```diff
@@ -461,6 +461,22 @@ func (f FeatureID) ToExpressionID() ExpressionID {
 	panic("Not an expression")
 }
 
+type Reference interface {
+	Source() FeatureID
+	Index() (int, error)
```

Comment:
Rather than requiring all References to have an Index(), which only applies to points on a path right now, is there an argument for allow sub interfaces of Reference, ie:

```
type Reference interface {
    Source() FeatureID
}

type PositionReference interface {
    Reference
    Index() int
}
```

Reply:
ye i like it my constructor if was ugly anyways